core/trino-main/src/main/java/io/trino/sql/planner/QueryPlanner.java

```diff
@@ -570,12 +572,24 @@ public PlanNode plan(Delete node)
                         handle,
                         Optional.empty(),
                         tableMetadata.table(),
-                        paradigmAndTypes),
+                        paradigmAndTypes,
+                        findSourceTableHandles(projectNode)),
                 projectNode.getOutputSymbols(),
                 partitioningScheme,
                 outputs);
     }
 
+    private static List<TableHandle> findSourceTableHandles(PlanNode startNode)
```

Comment:
nit: I would move after `plan(Update node)` since it is called from the method.

Reply:
I'm guessing you meant `io.trino.sql.planner.QueryPlanner#createMergePipeline` - moved it below this method.